src/azure-cli-core/azure/cli/core/__init__.py

```diff
@@ -277,7 +277,6 @@ def load_arguments(self, command):
                 loader.load_arguments(command)  # this adds entries to the argument registries
                 self.argument_registry.arguments.update(loader.argument_registry.arguments)
                 self.extra_argument_registry.update(loader.extra_argument_registry)
-                loader._update_command_definitions()  # pylint: disable=protected-access
```

Comment:
Has to be relocated so the CMD_TBL_LOADED event can be called and processed before the arguments are updated.